pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,304 @@
+        :type diagonal_motion: bool
+        :param max_time: Maximum time allowed for planning, in seconds.
+        :type max_time: float
+        """
+
+        self.world = world
+        self.max_time = max_time
+        self.resolution = resolution
+        self.distance_metric = distance_metric
+        self.diagonal_motion = diagonal_motion
+        self.inflation_radius = inflation_radius
+
+        self.goal = None
+        self.start = None
+        self.planning_time = 0
+        self.num_nodes_expanded = 0
+        self.candidates = PriorityQueue()
+        self.latest_path = Path()
+        # This dictionary keeps track of the parent node for each expanded node
+        # eg : self.parent_of[(10,10)] = (9,10)
+        self.parent_of = {}
+        # This dictionary keeps track of the cost till the particular node
+        # eg : self.cost_till((10, 10)) = 20
+        self.cost_till = {}
+
+        self._set_actions()
+        self._set_heuristic()
+        self._set_occupancy_grid()
+
+    def _set_occupancy_grid(self):
+        """
+        Generates occupancy grid of specified configuration
+        """
+        ts = time.time()
+        self.grid = occupancy_grid_from_world(
+            self.world,
+            resolution=self.resolution,
+            inflation_radius=self.inflation_radius,
+        )
+        self.grid_generation_time = time.time() - ts
+
+    def _set_actions(self):
+        """
+        Generates the actions available
+        """
+        orthogonal_distance = 1
+        diagonal_distance = math.sqrt(2)
+        self.actions = {
+            "left": {"cost": orthogonal_distance, "action": (-1, 0)},
+            "right": {"cost": orthogonal_distance, "action": (1, 0)},
+            "up": {"cost": orthogonal_distance, "action": (0, 1)},
+            "down": {"cost": orthogonal_distance, "action": (0, -1)},
+            "left_up": {"cost": diagonal_distance, "action": (-1, 1)},
+            "left_down": {"cost": diagonal_distance, "action": (-1, -1)},
+            "right_up": {"cost": diagonal_distance, "action": (1, 1)},
+            "right_down": {"cost": diagonal_distance, "action": (1, -1)},
+        }
+        keys = list(self.actions.keys())
+        self.selected_actions = keys if self.diagonal_motion else keys[:4]
+
+    def _set_heuristic(self):
+        """
+        Sets the heuristic
+        """
+        euclidean = lambda p1, p2: math.sqrt(
+            (p1[0] - p2[0]) ** 2 + (p1[1] - p2[1]) ** 2
+        )
+        manhattan = lambda p1, p2: abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])
+
+        if self.distance_metric == "euclidean":
+            self._heuristic = euclidean
+        elif self.distance_metric == "manhattan":
+            if not self.diagonal_motion:
+                warnings.warn("Manhattan over estimates without diagonal motion")
+            self._heuristic = manhattan
+        elif self.distance_metric == "none":
+            self._heuristic = lambda p1, p2: 0
```

Comment:
If a user enters a garbage value, they won't have a default behavior or a warning or error or anything. They just get an uninitialized `self._heuristic`, right?

Reply:
Should  this default to "none" and print a warning?

Reply:
that seems good to me!

Reply:
Also, what do you think about renaming the `distance_metric` to `heuristic` instead.
I think it makes more sense here.

Reply:
I don't have a strong preference on this one, so feel free to update it!